### PR TITLE
Update GrblDriver.java

### PR DESCRIPTION
**Pull request by cri-s**
Grbl Driver update

### CODE REVIEW

**Comment by vonnieda** [comment]
Can you please describe what the purpose of the changes are?

**Comment by cri-s** [comment]
The purpose of this change is just update for Grbl Driver.
The status of the old Driver:
Homing not work, what instead of homing  work is just Go to pos 0 0 0
Only custom Grbl version work, no official version, even older works.
No support for switching to mm unit, it just use the configured unit.
Stop button don't interrupt anything, in fakt it emit nothing on the
driver side.
No support for 3 Axis Grbl

This modification cleanup with this issues and support all versions of Grbl.


2015-02-02 16:26 GMT, Jason von Nieda <notifications@github.com>:

**Comment by cri-s** [comment]
I have just seen there are unicode probles, i have just copy and paste this code from editor.
I must manually check in, so wait pls because the code compiles and runs without problems,
but here sometimes ): is added.

File: src/main/java/org/openpnp/machine/reference/driver/GrblDriver.java [hunk 1]
```diff
@@ -51,92 +49,84 @@
 /**
  * TODO: Consider adding some type of heartbeat to the firmware.  
  */
+
 public class GrblDriver extends AbstractSerialPortDriver implements Runnable {
+
```
Comment (vonnieda) [inline src/main/java/org/openpnp/machine/reference/driver/GrblDriver.java]: These two variables do not appear to be used. Can they be removed?

File: src/main/java/org/openpnp/machine/reference/driver/GrblDriver.java [hunk 2]
```diff
@@ -51,92 +49,84 @@
...
             @Override
             public void configurationComplete(Configuration configuration)
                     throws Exception {
                 connect();
             }
-        });
-	}
-	
-	@Override
-	public void actuate(ReferenceActuator actuator, boolean on)
```
Comment (vonnieda) [inline src/main/java/org/openpnp/machine/reference/driver/GrblDriver.java]: Why was this removed?

File: src/main/java/org/openpnp/machine/reference/driver/GrblDriver.java (hunk 1 above)
Comment (cri-s) [inline src/main/java/org/openpnp/machine/reference/driver/GrblDriver.java]: do you mean
+//	private static final double minimumRequiredVersion = 0.75;
+//	private double connectedVersion;

This are commented out. For now i prefer leaving it as is until
jogging is implemented, because when implementing
Jogging it can be possible that it is advisable to limit version numbering.
I don't have checked it for now.
On jogging, it can be advisable to limit only firmware builds that
have lookahead builded in
for not decellerating and stop during moves or direction/speed changes.

Comment (vonnieda) [inline src/main/java/org/openpnp/machine/reference/driver/GrblDriver.java]: I was referring to vacPumpPin and invertVacpump. If they are not needed then they should be removed.

File: src/main/java/org/openpnp/machine/reference/driver/GrblDriver.java [hunk 3]
```diff
@@ -51,92 +49,84 @@
...
+            sb.append(String.format(Locale.US, "X%2.4f ", x));
+        }
+        if (!Double.isNaN(y) && y != this.y) {
+            sb.append(String.format(Locale.US, "Y%2.4f ", y));
+        }
+        if (!Double.isNaN(z) && z != this.z) {
+            sb.append(String.format(Locale.US, "Z%2.4f ", z));
+        }
+        if (!Double.isNaN(c) && c != this.c) {
+            sb.append(String.format(Locale.US, "E%2.4f ", c));
```
Comment (vonnieda) [inline src/main/java/org/openpnp/machine/reference/driver/GrblDriver.java]: This should reference the C axis, not E.

**Comment by vonnieda** [comment]
Changes look good overall. I added a few comments where I'd like to see the code improved before merging.

In the future, please try to limit pull requests to a single feature, fix or change. Having many different changes in a pull request makes it very hard to verify.
